### DISCUSSION TITLE
의존성이란 무엇이고, 의존성이 높으면 어떤 점이 안좋나요?

### DISCUSSION THREAD

**Discussion by kunsanglee** [2024-woowacourse-study/level-interview#66] [OOP]
.

**Comment by J-I-H-O** [comment]
의존성은 하나의 모듈(클래스, 패키지, 라이브러리 등)이 다른 모듈에 의존하는 관계를 나타냅니다.

의존성이 높으면 모듈 간의 결합도가 증가합니다. 하나의 모듈을 수정할 때 다른 모듈도 함께 수정해야 할 가능성이 높아지고, 이에 따라 유지보수가 어려워지게 됩니다.
또한 의존성이 높으면 단위 테스트를 작성하기 어려워지고, 다른 모듈에 강하게 의존하는 코드는 해당 모듈과 분리되어 재사용하기 어렵습니다.

**Comment by 3Juhwan** [comment]
의존성은 하나의 모듈이 다른 모듈에게 의존하는 관계를 나타냅니다. 즉, 하나의 모듈이 다른 모듈의 기능이나 자원을 사용하거나 호출하는 것을 의미합니다. 

의존성이 높다는 것은 한 모듈이 다른 모듈에게 많은 기능이나 자원에 의존한다는 것을 의미합니다. 이 경우, 한 모듈의 변경이 다른 모듈에 큰 영향을 미치게 되며, 시스템의 유연성과 확장성이 저하될 수 있습니다. 모듈 간의 결합도가 증가하여 유지보수하기 어려운 코드가 발생합니다. 해당 모듈을 사용하기 위해 의존성을 갖는 모든 모듈에 의존성이 발생하여, 테스트와 코드 재사용이 어렵습니다.
